dashboard/src/actions/catalog.ts

```diff
@@ -60,16 +62,44 @@ const actions = [
 
 export type ServiceCatalogAction = ActionType<typeof actions[number]>;
 
+function isEmptyDeep(obj: any): boolean {
+  if (typeof obj === "number") {
+    // isEmpty(number) is true but it's not empty
+    return false;
+  }
+  if (typeof obj === "object" && !_.isEmpty(obj)) {
+    // Check if nested objects are empty
+    // If some of the keys are not empty the result is not empty
+    return !Object.keys(obj).some(k => {
+      return !isEmptyDeep(obj[k]);
+    });
+  }
+  return _.isEmpty(obj);
+}
+
+function removeEmptyFields(obj: object, schema?: JSONSchema6) {
+  const res = { ...obj };
+  Object.keys(res).forEach(k => {
+    // Delete the key if it's empty and it's marked as optional by the schema
+    if (isEmptyDeep(res[k]) && schema && schema.required && schema.required.indexOf(k) === -1) {
+      delete res[k];
+    }
+  });
+  return res;
+}
+
 export function provision(
   releaseName: string,
   namespace: string,
   className: string,
   planName: string,
   parameters: {},
+  schema?: JSONSchema6,
 ): ThunkAction<Promise<boolean>, IStoreState, null, ServiceCatalogAction> {
   return async dispatch => {
     try {
-      await ServiceInstance.create(releaseName, namespace, className, planName, parameters);
+      const filteredParams = removeEmptyFields(parameters, schema);
```

Comment:
maybe we do not need to pass schema all the time since it is actually optional and with default value to `JSONSchema6`? Or do we support other schemas?

Reply:
I don't understand what you mean here

Reply:
My bad, I was understanding schema like the schema type and thought that the default value was `JSONSchema6`. When what it actually does it to receive an instance of jsonschema.